### PR TITLE
Add sdevent adapter

### PR DESCRIPTION
**Pull request by Oipo**

### CODE REVIEW

**Comment by michael-grunder** [comment]
Thanks for the PR, I'll go over it this weekend.

File: adapters/libsdevent.h [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#ifndef __HIREDIS_LIBSDEVENT_H__
+#define __HIREDIS_LIBSDEVENT_H__
```
Comment (michael-grunder) [inline adapters/libsdevent.h:2]:
```suggestion
#ifndef HIREDIS_LIBSDEVENT_H
#define HIREDIS_LIBSDEVENT_H
```

Leading underscores are technically an ISO violation.  We have other headers that violate the rule as well, which I am planning on getting to. :smile:

File: examples/example-libsdevent.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        return 1;
+    }
+
+    redisLibsdeventAttach(c,event);
+    redisAsyncSetConnectCallback(c,connectCallback);
+    redisAsyncSetDisconnectCallback(c,disconnectCallback);
+    redisAsyncSetTimeout(c, (struct timeval){ .tv_sec = 1, .tv_usec = 0});
+
+    /*
+    In this demo, we first `set key`, then `get key` to demonstrate the basic usage of libuv adapter.
```
Comment (michael-grunder) [inline examples/example-libsdevent.c:69]:
```suggestion
    In this demo, we first `set key`, then `get key` to demonstrate the basic usage of libsdevent adapter.
```

File: examples/example-libsdevent.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+
+#include <hiredis.h>
+#include <async.h>
+#include <adapters/libsdevent.h>
+
+void debugCallback(redisAsyncContext *c, void *r, void *privdata) {
+    (void)privdata; //unused
```
Comment (michael-grunder) [inline examples/example-libsdevent.c:11]:
```suggestion
    (void)privdata;
```

**Review by bjosv — COMMENTED** [review]
Nice, just a small nit..

File: adapters/libsdevent.h [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+static void redisLibsdeventAddWrite(void *userdata) {
+    redisLibsdeventEvents *e = (redisLibsdeventEvents*)userdata;
+
+    if (e->flags & EPOLLOUT) {
+        return;
+    }
+
+    e->flags |= EPOLLOUT;
+
+    if(e->flags & EPOLLIN) {
```
[the remark this answers is not in the document]
Comment (Oipo) [reply adapters/libsdevent.h]: I changed this in the two files I touched in this PR, but it seems there are more instances throughout the code (e.g. `macosx.h`, `test.c` and `net.c` f.e.

**Comment by Oipo** [comment]
Thanks for the review, updated the code.

**Comment by michael-grunder** [comment]
Merged, thanks!

**Pull request merged**